jena-db/jena-tdb2/src/main/java/org/apache/jena/tdb2/sys/DatabaseOps.java

```diff
@@ -195,6 +207,19 @@ public static void compact(DatasetGraphSwitchable container) {
             LOG.debug(String.format("Compact %s -> %s\n", db1.getFileName(), db2.getFileName()));
 
             compact(container, loc1, loc2);
+
+            if ( !Sys.isWindows && shouldDeleteOld ) {
```

Comment:
As above. Don't need to check for windows - it is only tests that are really affects by this.

If you think it is better to alter the user, a log message could be issued.

Reply:
fixed